types/recharts/index.d.ts

```diff
@@ -25,7 +26,7 @@ export type TooltipFormatter = (value: string | number | Array<string | number>,
                                 entry: TooltipPayload, index: number) => React.ReactNode;
 export type ItemSorter<T> = (a: T, b: T) => number;
 export type ContentRenderer<P> = (props: P) => React.ReactNode;
-export type DataKey = string | number | ((dataObject: any) => number | [number, number]);
+export type DataKey = string | number | ((dataObject: any) => number | [number, number] | null);
```

Comment:
I'm a bit confused about this change. Could you clarify why returning null here is desirable?

Reply:
The DataKey can be a function which maps data points to their y values on the graph. Not all data points will necessarily have a y-value, e.g. if you have two different data series with slightly different ranges.

The example [here](http://recharts.org/en-US/examples/LineChartConnectNulls) shows that null values are not unexpected, yet the DataKey function was previously not allowed to return null

Reply:
DataKey is not mandatory, and recharts uses isNil hence it can accept both null and undefined, that is why I approved the changes.

The example you show (as far is I can see) does not have an empty DataKey though. It has an empty value for an axis, which is a different thing.

Reply:
That's helpful, thank you!